src/pudl/output/eia860.py

```diff
@@ -136,11 +138,168 @@ def plants_eia860(pudl_engine, start_date=None, end_date=None):
+    fills in the old years with BA codes using two main methods:
+
+    * Backfilling from the oldest reported BA code (via :meth:`pd.bfill`)
+    * Using the most consistent reported value
+
+    We add a column to represent each of these two methodologies via
+    :func:`make_backfilled_ba_code_column` and :func:`make_consistent_ba_code_column`.
+
+    We know that the BA codes do change over time and are incorrectly reported at times.
+    Because of these two facts, we cann't simple :meth:`pd.fillna` with either the
+    backfilled or most consistent value. This function employs some specific filling in
+    methodologies based on an investigation of the data. Here is a description of each
+    of various stages of filling in which involves some bespoke cleanup:
+
+    * if the backfilled code and the most consistent code are the same, use either!
+    * assign BA code ``PACW`` (``PacifiCorp - West``) when ``PACW`` is the most
+      consistent BA code and the ``state`` is within ``PACW``'s territory
+    * similarity, assign BA code ``PACE`` (``PacifiCorp - East``) when ``PACE`` is
+      the most consistent BA code and the ``state`` is within ``PACE``'s territory
+    * use the backfilled code for plants that have ``SWPP`` (``Southwest Power Pool``)
+      as their most consistent BA code because we know ``SWPP`` accumulated many smaller
+      balancing authorities
+    * use the backfilled code for plants that have a differnt backfilled and most
+      consistent code, but the most consistent code is has a consistency rate (how
+      frequent the most consistent BA code is reported / the total BA codes reported)
+      is less than 80%. This is meant to find plants where there are multiple years of
+      an older BA code before it switches - assuming a BA code being reported for
+      multiple years is not a reporting error. TODO: make this one more explicit.
+
+    Args:
+        plants: table of annual plant attributes, including ``balancing_authority_code_eia``
+    """
+
+    def log_current_ba_code_nulls(plants: pd.DataFrame, method_str: str) -> None:
+        """Internal function to log progress on fillin in BA codes.
+
+        Args:
+            plants: the current plants table to check
+            method_str: A description of the method employed. This will be inserted into
+                the log.
+        """
+        currently_null_len = len(plants[plants.balancing_authority_code_eia.isnull()])
+        logger.info(
+            f"Filled BA codes where {method_str}. "
+            f"Currently {currently_null_len/len(plants):.1%} of records ({currently_null_len}) with no BA codes"
+        )
+
+    plants = make_backfilled_ba_code_column(plants, by_cols=["plant_id_eia"]).pipe(
+        make_consistent_ba_code_column
+    )
+    log_current_ba_code_nulls(plants, method_str="no treatment had been applied")
+    # when the backfilled code and the static code are the same, use either result
```

Comment:
"...use the most consistent code."

I know it's supposed to be the same. But saying what we're actually doing can be helpful in debugging if something is weird.

Reply:
this is the first log. nothing has been filled in yet.

Reply:
Sorry I was referring to the comment line, where it says "use either result"

Reply:
i disagree here. the code very clearly is using the most consistent. but i think for an inline comment specifically it is helpful to know that we could use either.